### DISCUSSION TITLE
NexoPOS 5.0.0 Beta 1

### DISCUSSION THREAD

**Discussion by Blair2004** [Blair2004/NexoPOS#1600] [Releases]
We're releasing today a beta version for version 5 of NexoPOS. We want this release to encourage users to discover what would be the new feature of NexoPOS. Those who are interested can test the cloud print options which is also about to be released along with that version.

Note that this is not a stable version and shouldn't be used with v4 modules. You shouldn't also use this update on a production environment. We remain available for any bug reports and suggestions.

Changelog:
========
- Ensure compatibility with php artisan serve https://github.com/Blair2004/NexoPOS/issues/1585
- Added: migration to fix wrong purchase prices https://github.com/Blair2004/NexoPOS/issues/1522
- Fixed: product with modified price being set to 0 after loading
- Added: default fields for quick products
- Fixed: issue while opening order with the changed price
- Fixed: tax computation of reopened order
- Added: product tax on receipt if available
- Fix https://github.com/Blair2004/NexoPOS/issues/1558 : Displaying procurement for providers wasn't limited to that provider
- Removed: banned feature
- Added: new event ProductBeforeUpdatedEvent
- Refactored: user authentication
- Updated: use "post()" instead of "all()" on Request
- Refactored: Crud methods
- Added: compute category count when product is updated.
- Added: new product scope
- Added: register reset fields
- Added: a different message for updating orders
- Fixed: Added factory for filling empty email and username
- Fixed: nsLodingPopupVue not defined
- Fixed: undeclared variable used on the template
- Added: force check if the system is installed
- Updated: refactored Setup.php service
- Fixed: getPlainData to only use POST inputs
- Fixed: test to use carbon "toDateTimeString()"
- Fixed: yearly report not recomputing
- Fixed: order types not used on the demo
- Fixed: sale report row background color
- Removed: unnecessary console.log()
- Added: chunking vendors
- Added: new background color for input on the crud component
- Fixed: wrong attribute use on customer account history
- Added: a new action hook for the header (to inject content on the header)
- Fixed: $ no longer supported before braces on PHP 7.2
- Added: new Popup implementation
- Added: multiselect field to close on blur/click outside
- Updated: changed active payment implementation on payment popup
- Updated: ns-link no longer uses router-link
- Updated: order details to display first_name & last_name
- Updated: Popup implementation to use static method "show"
- Fixed: using localization on customer popup
- Updated: changed filter name from ns-dashboard-header to ns-dashboard-heade-file
- Added: Link to edit customers from the order details popup
- Refactored: Crud Components
- Updated: Error Page For Not Allowed Exception
- Added:  address fields generator
- Updated: changing "destroyed" to "unmounted" for vue3 compatibility
- Updated: Popup no longer creates anonymous vue  instances
- Fixed: XSS flaw
- Fixed: Permission Check Breach
- Fixed: added permission on Coupon CRUD
- Removed: deprecated api routes
- Updated: renamed method used on console
- Added: new not grouped scope to Product model
- Updated: Telescope configuration
- Updated: WithOrderTest.php trait
- Fixed: removing merge conflict text
- Fixed: https://github.com/Blair2004/NexoPOS/issues/1220
- Added: hook for product on default receipt
- Added: dependency check for providers
- Added: improved options saving
- Added: ban customers and restrict from POS (implements https://github.com/Blair2004/NexoPOS/discussions/1325)
- Added: search filter for modules
- Added: migration for the module has a --down argument to undo database changes.
- Added: module details command shows the API file + migrations.
- Removed: searchable is no longer a valid argument. A product can only be either on sale or not.
- Added: new header buttons can be added to CRUD component.
- Added: crud-footer is now set to only be an action.
- Added: permission has 2 new attributes that are fillable: (name & description).
- Added: log notification dispatched to unassigned roles.
- Fixed: Customer factory now generates customer properly.
- Fixed: Options values are now better parsed.
- Added: defineComponent & defineAsyncComponent available on global context (window).
- Remove: slot from vue components
- Updated: stored option to detect an array
- Fixed: an issue that might only affect old installations
- Added: clear temporary files
- Added: new filesystem disks
- Fixed: bug defining option values
- Added: new casts
- Updated: module generation command
- Added: module delete command
- Updated: Coupon Order History
- Added: raw property for casts crud entry properties
- Added: new module events
- Fixed: options saving and retrieving with an array
- Updated: modules service with test cases
- Fixed: seeders for the providers
- Fixed: DateTime on sale report
- Added: keydown support for search field on crud component
- Updated: coupon test to ensure history is created after a valid usage
- Fixed: double submit event on vue components.
- Fixed: handling cases when customers, groups, etc is not provided while creating coupons.
- Added: reward relationship to CustomerReward model.
- Removed: driver role
- Updated: change return type for "getForm" method on SettingsPage service.
- Removed: duplicate permission
- Update: removed deprecated response implementation
- Added: restrict coupons by customers and customers group
- Added: 4 new events for enabling and disabling modules.
- Fixed: use first_name instead of name.
- Added: URL to order type options.
- Added: support for tabs on settings Vue component.
- Update: refactoring code.
- Updated: moving tests on phpunit.ci.xml.
- Added: new casts for OrderCrud.php.
- Added: autoload for CouponOrderHistory.
- Updated: Autogenerate username, email if necessary on CustomerCrud.
- Updated: ensure the export feature can export around +700 columns.
- Updated: refactored extractCrudValitation to extractValidation.
- Updated: provide a better global scope for Customer.php model.
- Updated: adding padding and fixing colors (light) for the crud table.
- Fixed: ns-best-customers to use "first_name" in place or "name"
- Fixed: test
- Added: new casts
- Added: new tabs on the user profile
- Removed: deprecated hook
- Updated: changing "name" and "surname" to "first_name" and "last_name"
- Updated: ensure "Customer" model points to user but only loads user with customer role.
- Updated: dashboard menus for a bette readability
- Delete: unused tables
- Fixed: DevShortcut command
- Added: coupon usage history
- Added: crud autoload
- Deprecated: addAction on CrudEntry, use "action" instead
- Added: dispatch user activation
- Added: middleware to destroy sessions when app is not installed
- Added: named arguments support on CrudService::table method
- Added: new roles (customers & drivers)
- Fixed: avatar image to ignore null src URL
- Updated: export classes from ns-menu and ns-submenu
- Updated: crud.blade.php generator to support PHP 8.1 syntax
- Added: new tests for coupon usage
- Added: new doctor command to recompute cash flow
- Added: new template tag to display order type
- Refactoring API Routes (will cause massive compatibility with v4.x modules
- Added: Profile settings
- Refactoring User model
- Refactoring Various Crud Class
- Added: Using Laravel Gate for authorization
- Added: Crud to support cast
- Updated: renaming Users class
- Updated: improve performance by loading widgets only where it's necessary
- Added: new permission per widgets
- Added: search field for permission
- Fixed: ensures widgets are always provided as an array
- Deprecated: dashid and alternative dashboard
- Added: support for permission on widgets
- Fixed: design issues
- Added: profile widget & other metric widgets
- Fixed: math accuracy when no precision is defined on the currency.
- Added: new math service
- Refactored: RecomputeCashFlowForDate
- Update: UpdateService to ignore non-php files
- Fixed: missing components on update.ts
- Added: recurrence fields
- Added: Warning for recurring salary
- Added: new date range field
- Fixed: missing user groups on expenses fields
- Fixed: error while exporting an empty table
- Added: calendar component
- Added: vue 3 components previewer
- Fix: https://github.com/Blair2004/NexoPOS/issues/1133 prevent discount usage on payment popup.
- First Build Vite+Vue3
- Removed Vue2 Builds
- Added: missing fonts
- Refactored: login request subscription
- Added: dependency protection on build deletion (customers).
- Added: default type for the unsupported product type.
- Added: adjust stock for order product if a reference exists.
- Added: new cash register action
- Fix: https://github.com/Blair2004/NexoPOS/issues/1100 double records for sales
- Updated: refactoring
- Fixed: deleting own account from Crud List
- Added: new hooks to inject content on top and bottom of a title
- Added: a new test for checking users
- Added: implementing new Math calculator
- Added: a new test case
- Updated: CurrencyService class
- Added: compute product taxes command
- Added: product taxes on POS
- Added: new test cases
- Removed: deprecated code
- Refactored: taxes on products and order products
- Fixed: https://github.com/Blair2004/NexoPOS/issues/1071 order not updated after a refund
- Fixed: design issues
- Fixed: quick product validation failing
- Fix: https://github.com/Blair2004/NexoPOS/issues/1070 Customers Not selected by default
- Updated: module listing command
- Added: automatic page output action
- Added: global nsSelectPopup & nsPOSLoadingPopup
- Added: silent and normal printing
- Added: name on web routes
- Fixed: a wrong method called on ns-table-rows.ts
- Fixed: deprecated issue with str_replace
- Fixed: option typo "only_paid_ordes" => "only_paid_orders"
- Fixed: design issue with notices
- Fixed: list background color on hover
- Added: a new test for timezones
- Deleted: incompatible timezone
- Added: float casting
- Added: option to mark a procurement as paid
- Added: a new test case for procurements
- fix: unable to load style on receipts
- feature: new events to catch product inventory update
- fix https://github.com/Blair2004/NexoPOS/issues/1027: translation on the POS cart
- updated: translation
- Updated: improving rebranding
- Fixed: Products set as hidden shouldn't be visible on POS [fix https://github.com/Blair2004/NexoPOS/issues/1015]
- Added: code parser while activating module
- Updated: compatibility with RxJs.
- Added: a way to disable all modules.
- Fixed: fields hidden while choosing a regular product (quick product)
- Updated: customer seeding.
- Updated: delete the previous module before uploading the new one.
- Fixed: Customer Popup Issue [fix https://github.com/Blair2004/NexoPOS/issues/994]
- Fixed: Add Billing Details By Default
- Added: pint project
- Updating Syntax for PHP 8.1
- Fixed: category total items count being refreshed
- Added: a new strategy for multi-tenancy
- Fixed: ensure customer with no email can be created
- Added: catch exception from custom actions.
- Fixed: 404 error while trying to create provider product https://github.com/Blair2004/NexoPOS/issues/941
- fix https://github.com/Blair2004/NexoPOS/issues/944 barcode reading issues
- Added: cleaning code
- Fixed: bug while selling order (purchase_amount)
- Fixed: disabling casting for procurement
- Fixed: testing issues
- Added: new fields for orders that support installments
- Fixed: get the last purchase price for defined products
- Fixed: remove forced option on artisan
- Fixed: issue with products images
- Fixed: increased range for users
- Added: command generator for modules
- Moving: migration files
- Fixed: increasing cash register ranges
- Fixed: ensure the order purchase price is updated
- Added: new doctor method to fix orders products
- Fixed: Removing unnecessary layaway restriction
- Added: new event & Listeners
- Fixed: create installment entries when it's necessary
- Fixed: reduce database queries on the user model

<hr /><em>This discussion was created from the release <a href='https://github.com/Blair2004/NexoPOS/releases/tag/v5.0.0-beta-1'>NexoPOS 5.0.0 Beta 1</a>.</em>

**Comment by lim10soft** [comment]
Nexopos Beta gives an error while installing and stays on the license side in Nexo Print Server 5.

**Reply by Blair2004** [reply]
Hey, Thank you for reporting, there are few things not working as smooth as expected we're working actively on that.

**Comment by ARMBouhali** [comment]
Are you planning to migrate from laravel mix to vite for v5 ?

**Reply by Blair2004** [reply]
Hi there, NexoPOS v5 uses Vite

**Comment by lim10soft** [comment]
The error I get when I install composer > Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Illuminate\Encryption\MissingAppKeyException

  No application encryption key has been specified.

  at vendor\laravel\framework\src\Illuminate\Encryption\EncryptionServiceProvider.php:79
     75▕     protected function key(array $config)
     76▕     {
     77▕         return tap($config['key'], function ($key) {
     78▕             if (empty($key)) {
  ➜  79▕                 throw new MissingAppKeyException;
     80▕             }
     81▕         });
     82▕     }
     83▕ }

  i   Your app key is missing:
      https://laravel.com/docs/master/installation#configuration

  1   vendor\laravel\framework\src\Illuminate\Support\helpers.php:306
      Illuminate\Encryption\EncryptionServiceProvider::Illuminate\Encryption\{closure}()
  2   vendor\laravel\framework\src\Illuminate\Encryption\EncryptionServiceProvider.php:81

Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1   I cannot proceed to the installation phase I am using Laragon php 8.1.19

**Comment by athahersirnaik** [comment]
When is v5 going to be released. Can you create milestones, so that developers can help with PRs
